src/components/map/Map.tsx

```diff
@@ -74,6 +81,37 @@ export const Map = ({
     };
   }
 
+  // ref function to handle changes in the viewable items of the FlatList
+  // it sets the activeMarker state to the ID of the first viewable item
+  const onViewableItemsChanged = useRef(({ viewableItems }) => {
+    const firstViewableItem = viewableItems[0]?.item;
+    if (firstViewableItem) {
+      setActiveMarker(firstViewableItem.id);
+    }
+  }).current;
+
+  // useEffect to handle changes in activeMarker
+  // it animates the map to focus on the coordinates of the active marker
+  useEffect(() => {
+    if (activeMarker && refForMapView.current) {
+      const marker = locations?.find((loc) => loc.id === activeMarker);
```

Comment:
coming back to the marker hint of @d4v3d4vE, commenting it here as I cannot find a better place, but to be able to have a connected conversation:
we need that active marker with a different color to finish this PR.

Reply:
should be done in #921 now